### PR TITLE
Update MOT17 metrics and add MOT20 results

### PR DESCRIPTION
**Pull request by facundo-lezama**
This PR recalculates MOT Challenge's metrics, more specifically it updates MOT17 metrics and adds MOT20 metrics. The biggest change in performance comes from using ByteTrack's YOLOX object detection model to get the detections instead of using the original detections from MOT Challenge. This is a common practice in top-ranked tracking libraries since the original detections aren't precise. 

Results are reported on the `train` split since we don't have the ground truth for the `test` split. `Test` results must be obtained by uploading the tracking output to MOT Challenge. We are going to obtain them in a near future and update the results then. 

Changes in this PR include:
- Adding the camera motion feature to the motmetrics demo as we noticed that some sequences have camera movement.
- Adding MOT20 challenge to motmetrics demo.
- Using ByteTrack's YOLOX detections instead of the original ones.
- Simplify the motmetrics test to use the new predefined distances.
- Update README's MOT metrics.
  - MOT17 metrics show fewer rows. This is because originally we had 3 different sets of detections per video, but now that we find our own detections it doesn't make sense to process 3 times each video.

We decided not to change the motmetrics test to include MOT20 or the new detections since the overall result isn't important, we are only checking if everything remained the same. We also decided not to add camera motion to the test since this implied using the entire dataset with images and it would be too heavy to download in CI.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,32 +162,27 @@ If you are looking for a tracker, here are some other projects worth noting:
...
+
+|   MOT17 Train  |   IDF1  IDP  IDR  | Rcll  | Prcn  |  MOTA MOTP  |
+| :------------: | :---------------: | :---: | :---: | :---------: |
+| MOT17-02-FRCNN | 61.3% 63.6% 59.0% | 86.8% | 93.5% | 79.9% 14.8% |
+| MOT17-04-FRCNN | 93.3% 93.6% 93.0% | 98.6% | 99.3% | 97.9% 07.9% |
+| MOT17-05-FRCNN | 77.8% 77.7% 77.8% | 85.9% | 85.8% | 71.2% 14.7% |
+| MOT17-09-FRCNN | 65.0% 67.4% 62.9% | 90.3% | 96.8% | 86.8% 12.2% |
+| MOT17-10-FRCNN | 70.2% 72.5% 68.1% | 87.3% | 93.0% | 80.1% 18.7% |
+| MOT17-11-FRCNN | 80.2% 80.5% 80.0% | 93.0% | 93.6% | 86.4% 11.3% |
+| MOT17-13-FRCNN | 79.0% 79.6% 78.4% | 90.6% | 92.0% | 82.4% 16.6% |
```
Comment (aguscas) [inline README.md:175]: I'm not sure if the names should end with "-FRCNN" given that this suffix was used to specify the model but now we are using a custom detector

Comment (facundo-lezama) [reply README.md:175]: You're right, it doesn't make sense now.

File: demos/motmetrics4norfair/src/motmetrics4norfair.py [hunk 2]
```diff
@@ -114,35 +95,39 @@ def keypoints_distance(detected_pose, tracked_pose):
...
-        distance_threshold=distance_threshold,
-        detection_threshold=detection_threshold,
-        pointwise_hit_counter_max=pointwise_hit_counter_max,
-        hit_counter_max=hit_counter_max,
+        distance_function=iou,
+        distance_threshold=DISTANCE_THRESHOLD,
+        detection_threshold=DETECTION_THRESHOLD,
+        pointwise_hit_counter_max=POINTWISE_HIT_COUNTER_MAX,
+        hit_counter_max=HIT_COUNTER_MAX,
         filter_factory=FilterPyKalmanFilterFactory(),
```
Comment (javiber) [inline demos/motmetrics4norfair/src/motmetrics4norfair.py:111]: why are we still using filterpy? OptimizedFilter is faster and it _shouldn't_ affect metrics too much.

**Review by javiber — APPROVED** [review]
LGTM, just had one question regarding the filter

File: demos/motmetrics4norfair/src/motmetrics4norfair.py (hunk 2 above)
Comment (facundo-lezama) [reply demos/motmetrics4norfair/src/motmetrics4norfair.py:111]: I tried using `OptimizedKalmanFilterFactory`, but the metrics were a bit worse. We saw this when implementing `OptimizedKalmanFilterFactory` but we still haven't gotten to the bottom of it.

**Pull request merged**